src/coreclr/vm/appdomain.hpp

```diff
@@ -1223,8 +1223,8 @@ class BaseDomain
 
 #ifdef HOST_WINDOWS
     // MethodTable to `typeIndex` map. `typeIndex` is embedded in the code during codegen.
-    // During execution corresponding thread static data blocks are stored in `t_threadStaticBlocks`
-    // array at the `typeIndex`.
+    // During execution corresponding thread static data blocks are stored in `t_NonGCThreadStaticBlocks`
+    // and `t_GCThreadStaticBlocks` array at the `typeIndex`.
```

Comment:
>  Instead, the NonGC-thread statics can be stored on odd indices and GC-thread statics can be stored on even indices.

Maybe I am missing something, but those indices are generated from `TypeIDMap` and for non-gc type, for eg. we can get `2` and `3`. How will we force them to store in even index in that case? 

> We are going to waste slots for these types with this design

Thinking out loud, if one type dominates, we will end up in a situation that they will all get stored at one of the even/odd index (depending on what we decide) and that will grow the cache with bigger amount, leading to wasting more slots, right?

Reply:
> if one type dominates

It is more like if one type is used exclusively. For example, you can get 1000 non-GC statics first and then you hit one GC static. The one GC static will make us allocated 1000 empty GC static slots.

Real-world programs are very likely going to have an unlucky thread static of the other type, that pushes the number of slots of both types to be nearly the same.

Reply:
> you can get 1000 non-GC statics first and then you hit one GC static. The one GC static will make us allocated 1000 empty GC static slots.

Actually, now I realize your point. This is happening because I am reusing the `m_threadStaticBlockTypeIDMap` which is still giving me higher `typeIndex`. I will just have a separate `m_GCThreadStaticBlockTypeIDMap` I think to keep them totally separate.